### PR TITLE
Change model names

### PR DESCRIPTION
**Pull request by heplesser**
@akorgor @JesusEV With this PR, the model names are changed and tests and examples adjusted to the new names.
Note that the handwriting and the infinite loop example still report failure with differences from the first time step, so they still need to be adjusted.

### CODE REVIEW

**Comment by heplesser** [comment]
You can ignore the failing Dependency Review and macOS tests.

File: modelsets/eprop [hunk 1]
```diff
@@ -8,12 +8,12 @@ spike_generator
 step_rate_generator
 poisson_generator
 
-eprop_iaf_psc_delta
-eprop_iaf_psc_delta_adapt
-eprop_readout
+eprop_delta_bsshslm_2020
+eprop_delta_bsshslm_2020_adapt
```
Comment (akorgor) [inline modelsets/eprop]:
```suggestion
eprop_iaf_bsshslm_2020
eprop_iaf_bsshslm_2020_adapt
```

File: models/eprop_readout_bsshslm_2020.h [hunk 2]
```diff
@@ -171,25 +171,25 @@ See also
...
+.. listexamples:: eprop_readout_bsshslm_2020
 
 EndUserDocs */
 
-void register_eprop_readout( const std::string& name );
+void register_eprop_readout_bsshslm_2020( const std::string& name );
 
 /**
  * Class implementing a current-based leaky integrate readout neuron model with delta-shaped postsynaptic currents for
  * e-prop plasticity.
```
Comment (akorgor) [inline models/eprop_readout_bsshslm_2020.h]:
```suggestion
 * e-prop plasticity according to Bellec et al. (2020).
```

File: models/eprop_learning_signal_connection_bsshslm_2020.h [hunk 3]
```diff
@@ -110,13 +111,13 @@ Examples using this model
 
 EndUserDocs */
 
-void register_eprop_learning_signal_connection( const std::string& name );
+void register_eprop_learning_signal_connection_bsshslm_2020( const std::string& name );
 
 /**
  * Class implementing a synapse model transmitting secondary feedback learning signals for e-prop plasticity.
```
Comment (akorgor) [inline models/eprop_learning_signal_connection_bsshslm_2020.h]:
```suggestion
 * Class implementing a synapse model transmitting secondary feedback learning signals for e-prop plasticity according to Bellec et al. (2020).
```

**Review by akorgor — COMMENTED** [review]
For me, all tests pass.

I would suggest:

1. moving the `adapt` closer to the biological model name, that is `eprop_iaf_adapt_bsshslm_2020`.
2. moving the biological name to the end, so that all the models belonging to one variant are grouped together in the models directory, that is `eprop_bsshslm_2020_iaf` etc.
3. also parts of the archiving node will be probably specific to the variant (we can fix this in a later commit)
4. adjust l. 240 in `eprop_iaf_bsshslm_2020_adapt.h`
5. adjust l. 57 and 58  in `eprop_synapse_bsshslm_2020.h`
6. adjust l. 203, 2012, 343 of `eprop_synapse_bsshslm_2020.h`
7. adjust l. 62 `connector_base.h`
8. adjust l. 54 `test_eprop_bsshslm_2020_plasticity.py`
9. adjust l. 451 in eprop_readout_bsshslm_2020.h
10. adjust l. 110 in `eprop_learning_signal_connection_bsshslm_2020`
11. adjust l. 442 node.cpp
12. add "according to Bellec et al. (2020)" in l. 194 of `eprop_synapse_bsshslm_2020`

**Comment by JesusEV** [comment]
@heplesser Hi, I've tested this using both configurations – with and without Boost – for completeness. All examples from this branch have passed successfully. Is there, perhaps, any other difference in the tutorial scripts you are running?

**Comment by heplesser** [comment]
> For me, all tests pass.

That includes the examples, I suppose? As we discovered recently, the failing verification of the two examples on my computer is due to the use of the Clang compiler with its different random deviate generator (libc++ vs stdlibc++), which is used to randomise the membrane time constants:

https://github.com/jstapmanns/nest-simulator/blob/5e4ecf7c4cc7439bb827558176ee2becbeb142cb/pynest/examples/eprop_plasticity/eprop_supervised_regression_handwriting.py#L194

I think we concluded then that we wanted to remove the verification part from the examples. Could you do that?

> 1. moving the `adapt` closer to the biological model name, that is `eprop_iaf_adapt_bsshslm_2020`.

OK, done with next push.

> 2. moving the biological name to the end, so that all the models belonging to one variant are grouped together in the models directory, that is `eprop_bsshslm_2020_iaf` etc.

I would argue against that, because it breaks with our naming scheme where we put the author-year part at the end.

> 3. also parts of the archiving node will be probably specific to the variant (we can fix this in a later commit)

If so, we should do this later. For now, we should focus on the user-facing part.

> 4. adjust l. 240 in `eprop_iaf_bsshslm_2020_adapt.h`

Done.

> 5. adjust l. 57 and 58  in `eprop_synapse_bsshslm_2020.h`

Done.

> 6. adjust l. 203, 2012, 343 of `eprop_synapse_bsshslm_2020.h`

Done 

> 7. adjust l. 62 `connector_base.h`

Done

> 8. adjust l. 54 `test_eprop_bsshslm_2020_plasticity.py`

Done

> 9. adjust l. 451 in eprop_readout_bsshslm_2020.h

Done 

> 10. adjust l. 110 in `eprop_learning_signal_connection_bsshslm_2020`

Done

> 11. adjust l. 442 node.cpp

Done

> 12. add "according to Bellec et al. (2020)" in l. 194 of `eprop_synapse_bsshslm_2020`

Done

**Comment by akorgor** [comment]
I have now removed the verification from the tutorials in https://github.com/nest/nest-simulator/pull/2867/commits/b563ddd8ff04ef5e9f7aad0aff53f82e02b0edbb

**Comment by heplesser** [comment]
@akorgor @JesusEV I just merge the latest master into the PR to keep the diff small. Just small changes to model documentation and github action configurations.

**Pull request merged**